spec/index.bs

```diff
@@ -1894,6 +1900,111 @@ dictionary GPUVertexAttributeDescriptor {
 };
 </script>
 
+## {{GPUPipelineBase/getBindGroupLayout()|getBindGroupLayout}} ## {#get-bind-group-layout}
+
+<dl dfn-type="method" dfn-for="GPUPipelineBase">
+    : <dfn>getBindGroupLayout(unsigned long index)</dfn>
+    ::
+        <div algorithm="GPUProgrammablePipeline.getBindGroupLayout(unsigned long index)">
+            1. If |index| is greater or equal to {{GPULimits/maxBindGroups}} :
+
+                1. Throw a {{RangeError}}.
+
+            1. If |this|.getBindGroupLayout has already been called with |index|:
+
+                1. Return the same object.
```

Comment:
Do we need this in the algorithm? It doesn't remove the expectation that we'd get the same layout if we go through the rest of the algorithm again, so we might as well specify this as if it computes it every time.

Reply:
See discussion about [SameObject]. This explicit memoization means that the same JS object will be returned, with the same expando-properties.

Reply:
ugh, trying to imagine people not providing a pipeline layout, then creating a pipeline, asking it for a bind group layout repeatedly, and modifying the properties on it at the same time. yikes!